src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -617,6 +633,230 @@ BoundExpression convertArrayElement(BoundExpression element, TypeSymbol elementT
+                    typeArgs[0].Type,
+                    diagnostics);
+                var value = convertExpression(
+                    getPropertyValue(syntax, addElementPlaceholder, WellKnownMember.System_Collections_Generic_KeyValuePair_KV__get_Value, diagnostics),
+                    typeArgs[1].Type,
+                    diagnostics);
+
+                var statement = new BoundExpressionStatement(
+                    syntax,
+                    setValue(syntax, implicitReceiver, indexer, key, value))
+                { WasCompilerGenerated = true };
+
+                return element.Update(
+                    element.Expression,
+                    enumeratorInfo,
+                    element.ElementPlaceholder,
+                    addElementPlaceholder,
+                    addOrSetElement: statement,
+                    element.Type);
+            }
+
+            BoundExpression convertExpression(BoundExpression expr, TypeSymbol targetType, BindingDiagnosticBag diagnostics)
+            {
+                var useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                var conversion = Conversions.ClassifyImplicitConversionFromExpression(expr, targetType, ref useSiteInfo);
+                diagnostics.Add(expr.Syntax, useSiteInfo);
+                bool hasErrors = !conversion.IsValid;
+                if (hasErrors)
+                {
+                    GenerateImplicitConversionError(diagnostics, expr.Syntax, conversion, expr, targetType);
+                    // Suppress any additional diagnostics
+                    diagnostics = BindingDiagnosticBag.Discarded;
+                }
+                var result = CreateConversion(
+                    expr.Syntax,
+                    expr,
+                    conversion,
+                    isCast: false,
+                    conversionGroupOpt: null,
+                    wasCompilerGenerated: true,
+                    destination: targetType,
+                    diagnostics,
+                    hasErrors: hasErrors);
+                result.WasCompilerGenerated = true;
+                return result;
+            }
+
+            BoundExpression getPropertyValue(SyntaxNode syntax, BoundExpression receiver, WellKnownMember getMember, BindingDiagnosticBag diagnostics)
+            {
+                Debug.Assert(receiver.Type is { });
+                Debug.Assert(receiver.Type.OriginalDefinition.Equals(Compilation.GetWellKnownType(WellKnownType.System_Collections_Generic_KeyValuePair_KV), TypeCompareKind.AllIgnoreOptions));
+
+                var method = (MethodSymbol)GetWellKnownTypeMember(getMember, diagnostics, syntax: syntax);
+                if (method is null)
+                {
+                    // PROTOTYPE: Test missing property.
+                    return BadExpression(syntax, childNode: receiver);
+                }
+
+                method = method.AsMember((NamedTypeSymbol)receiver.Type);
```

Comment:
what does this do?

Reply:
simpler to just use `collectionType` instead of `(NamedTypeSymbol)receiver.Type`?

Reply:
`MethodSymbol MethodSymbol.AsMember(NamedTypeSymbol newOwner)` returns the member corresponding to `this` from constructed type `newOwner`.

`collectionType` is the target type, such as `Dictionary<int, int>`, but `receiver.Type` is the spread enumerator element type, such as `KeyValuePair<int, int>`.